loki/transform/transform_hoist_variables.py

```diff
@@ -341,6 +349,30 @@ def kernel_call_argument_remapping(self, routine, call, variables):
         new_args = tuple(v.clone(dimensions=None) for v in variables)
         return call.clone(arguments=call.arguments + new_args)
 
+    def kernel_inline_call_argument_remapping(self, routine, call, variables):
+        """
+        Append hoisted temporaries to inline function call arguments.
+
+        Parameters
+        ----------
+        routine : :any:`Subroutine`
+            The subroutine to add the variable declaration to.
+        call : :any:`InlineCall`
+            ProcedureSymbol to which hoisted variables will be added.
+        variables : tuple of :any:`Variable`
+            The tuple of variables to be declared.
+        """
+
+        if self.as_kwarguments:
+            new_args = dict((a.name, v.clone(dimensions=None)) for (a, v) in variables)
+            _call_clone = call.clone(kwparameters=call.kw_parameters)
+            _call_clone.kw_parameters.update(new_args)
```

Comment:
I find this pattern of in-place updating the call attribute a little weird, particularly since we expect to only add new kwargs to the end. Wouldn't something more similar to the other argument_remapping method be a little more intuitive?

Also, the clone call has a typo (`kwparameters` vs. `kw_parameters`), which suggests we don't have a test with existing `kw_parameters` :) Could you add one for that?

```suggestion
            kw_params = call.kw_parameters
            kw_params += tuple((a.name, v.clone(dimensions=None)) for (a, v) in variables)
            _call_clone = call.clone(kwparameters=dict(kw_params))
```